torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1583,7 +1861,26 @@ void ONNXShapeTypeInference(
         }
       }
     }
-    UpdateReliable(n);
+  }
+  UpdateReliable(n);
+
+  // For the node type that does nott have ComputeConstant logic, it may have
+  // reliable shape but its shape is not in ConstantValueMap. So we need this
+  // logic to update ConstantValueMap.
+  for (auto node_output : n->outputs()) {
+    if (ConstantValueMap::HasTypeReliable(node_output->debugName())) {
+      auto reliable =
+          ConstantValueMap::GetTypeReliable(node_output->debugName())
+              .value_or(false);
+      if (reliable && !ConstantValueMap::HasShape(node_output->debugName())) {
+        if (auto output_tensor_type = node_output->type()->cast<TensorType>()) {
```

Comment:
How about sequences and Tensors and other types? Should we handle those outputs?

Reply:
Add todo: ListType case